### PR TITLE
Added support for topics.regex

### PR DESCRIPTION
**Pull request by rozza**
Users now must set either the topics list or a topics regex as per: https://docs.confluent.io/current/installation/configuration/connect/sink-connect-configs.html

https://evergreen.mongodb.com/version/5e971c963e8e86625a2a9576

KAFKA-82

### CODE REVIEW

File: src/integrationTest/java/com/mongodb/kafka/connect/MongoSinkConnectorTest.java [hunk 1]
```diff
@@ -41,78 +41,59 @@
 class MongoSinkConnectorTest extends MongoKafkaTestCase {
 
     @Test
-    @DisplayName("Ensure simple producer sends data")
-    void testASimpleProducerSmokeTest() {
```
Comment (rozza) [inline src/integrationTest/java/com/mongodb/kafka/connect/MongoSinkConnectorTest.java:45]: This test was testing the producer worked and not the connector.  Its not needed so I removed it.

File: src/test/java/com/mongodb/kafka/connect/sink/SinkTestHelper.java [hunk 2]
```diff
@@ -40,12 +41,20 @@
 
     public static  Map<String, String> createConfigMap(final String json) {
         Map<String, String> map = createConfigMap();
-        Document.parse(json).forEach((k, v) -> map.put(k, v.toString()));
+        Document.parse(json).forEach((k, v) -> {
+            if (k.equals(TOPICS_REGEX_CONFIG)) {
+                map.remove(TOPICS_CONFIG);
```
Comment (rozza) [inline src/test/java/com/mongodb/kafka/connect/sink/SinkTestHelper.java:46]: By default TOPICS_CONFIG was always added in `createConfigMap` - this just ensures that its removed when the regex config is set.

File: src/test/java/com/mongodb/kafka/connect/sink/MongoSinkConfigTest.java [hunk 3]
```diff
@@ -140,14 +165,28 @@ void testMissingDatabaseName() {
...
+    @DisplayName("test topic regex with overrides")
+    void testTopicRegexWithOverrides() {
+        MongoSinkConfig cfg = createSinkConfig(format("{'%s': 't(.*)', '%s': 'otherDB', '%s': 'coll2', '%s': 'coll3'}",
+                TOPICS_REGEX_CONFIG, createOverrideKey("t2", DATABASE_CONFIG), createOverrideKey("t2", COLLECTION_CONFIG),
+                createOverrideKey("noMatch", COLLECTION_CONFIG)));
+        assertPattern("t(.*)", cfg.getTopicRegex().orElse(EMPTY_PATTERN));
+        assertEquals("myDB.topic", cfg.getMongoSinkTopicConfig("topic").getNamespace().toString());
+        assertEquals("otherDB.coll2", cfg.getMongoSinkTopicConfig("t2").getNamespace().toString());
+        assertThrows(ConfigException.class, () -> cfg.getMongoSinkTopicConfig("noMatch"));
+        assertThrows(ConfigException.class, () -> createSinkConfig(format("{'%s': 't(.*)', '%s': 'made up'}",
```
Comment (rozza) [inline src/test/java/com/mongodb/kafka/connect/sink/MongoSinkConfigTest.java:186]: Tests an invalid key projection, when using topic overrides and regex configs.

File: src/test/java/com/mongodb/kafka/connect/sink/MongoSinkConfigTest.java [hunk 4]
```diff
@@ -125,12 +130,32 @@ void testClientUri() {
...
+                        () -> createSinkConfig(TOPICS_REGEX_CONFIG, "topic-(.*)").getMongoSinkTopicConfig("alpha"))
+        );
+    }
+
+    @Test
+    @DisplayName("test topics and topic regex")
+    void testTopicsAndTopicRegex() {
+        Map<String, String> configMap = createConfigMap();
+        configMap.put(TOPICS_REGEX_CONFIG, ".*");
+        assertInvalid(TOPICS_CONFIG, configMap);
```
Comment (rozza) [inline src/test/java/com/mongodb/kafka/connect/sink/MongoSinkConfigTest.java:156]: Tests that you can't set both the Topic config and the regex.

**Review by jyemin — APPROVED** [review]
LGTM

**Pull request closed without merging**